lib/elastomer_client/client.rb

```diff
@@ -283,11 +271,6 @@ def request(method, path, params)
         # wrap Faraday errors with appropriate ElastomerClient::Client error classes
         rescue Faraday::Error::ClientError => boom
           error = wrap_faraday_error(boom, method, path)
-          if error.retry? && RETRYABLE_METHODS.include?(method) && (retries += 1) <= request_max_retries
```

Comment:
This is a bit odd, why do we only wrap client errors (i.e. 4xx status codes)? I would expect we would wrap everything and not just client errors.

See https://github.com/lostisland/faraday/blob/v0.17.6/lib/faraday/error.rb#L66-L68

Reply:
Looking at [handle_errors](https://github.com/github/elastomer-client/blob/main/lib/elastomer_client/client.rb#L425-L445) that doesn't handle timeouts either. Also, there isn't anything in the [connection](https://github.com/github/elastomer-client/blob/main/lib/elastomer_client/client.rb#L140-L167) that raises errors, so Faraday shouldn't be raising client errors.

The more I look into this, the more confused I get as to how this works.

Reply:
The only thing I can think of is that maybe [this meta-programming](https://github.com/github/elastomer-client/blob/main/lib/elastomer_client/client/errors.rb#L100-L109) is what enables retries. Though it shouldn't because it should skip anything that does not inherit from client error.